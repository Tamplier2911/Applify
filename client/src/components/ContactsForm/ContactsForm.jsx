// import "./ContactsForm.scss";
import React, { useState } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// components
import FormInput from "../FormInput/FormInput";
import TextInput from "../TextInput/TextInput";
import Button from "../Button/Button";

// JS Rendering CSS
import {
  ContactsFormContainer,
  ContactsFormTitle,
  ContactsFormItself
} from "./ContactsFormStyles";

// component constants
import contactsFormData from "../../utils/ComponentContactsFormConstants/componentContactsFormConstants";

const ContactsForm = ({ lang }) => {
  const [contactsInfo, setContactsInfo] = useState({
    name: "",
    email: "",
    message: ""
  });
  const { name, email, message } = contactsInfo;

  const onInputChange = e => {
    const { name, value } = e.target;
    setContactsInfo({ ...contactsInfo, [name]: value });
  };

  const onFormSubmit = e => {
    e.preventDefault();
    // PERFORM VALIDATION ON EMAIL
    console.log(contactsInfo);
  };

  const {
    contactsFormTitle,
    contactsFormNameField,
    contactsFormEmailField,
    contactsFormMsgField,
    contactsFormSubmitField,
    contactsFormMsgLength
  } = contactsFormData[lang];

  return (
    <ContactsFormContainer>
      <ContactsFormTitle lang={lang}>{contactsFormTitle}</ContactsFormTitle>
      <ContactsFormItself autoComplete="off" onSubmit={e => onFormSubmit(e)}>
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={name}
          label={contactsFormNameField}
          name="name"
          type="text"
          required
        />
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={email}
          label={contactsFormEmailField}
          name="email"
          type="email"
          required
        />
        <TextInput
          rows="6"
          onInputChange={e => onInputChange(e)}
          label={contactsFormMsgField}
          name="message"
          value={message}
          max="2000"
          //   placeholder="Don't be shy, share your ideas with me!"
          required
        />
        <Button type="submit" value={contactsFormSubmitField} />
      </ContactsFormItself>
      {contactsFormMsgLength + " "}
      {2000 - message.length}
    </ContactsFormContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps)(ContactsForm);
