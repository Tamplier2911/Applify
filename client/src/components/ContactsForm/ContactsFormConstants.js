const contactsFormData = {
  eng: {
    // contactsFormTitle:
    //   "Send me a message describing your problem, and I will get back to you shortly!",
    contactsFormTitle:
      "Send us a message describing your problem, and we will get back to you shortly!",
    contactsFormNameField: "Name",
    contactsFormEmailField: "Email",
    contactsFormMsgField: "Message",
    contactsFormSubmitField: "Send",
    contactsFormMsgLength: "Message length:",
  },
  rus: {
    // contactsFormTitle:
    //   "Отправьте нам сообщение с описанием вашей проблемы, и я свяжусь с вами в ближайшее время!",
    contactsFormTitle:
      "Отправьте нам сообщение с описанием вашей проблемы, и мы свяжемся с вами в ближайшее время!",
    contactsFormNameField: "Имя",
    contactsFormEmailField: "Эл. адрес",
    contactsFormMsgField: "Сообщение",
    contactsFormSubmitField: "Отправить",
    contactsFormMsgLength: "Длина сообщения:",
  },
  ukr: {
    // contactsFormTitle:
    //   "Надішліть мені повідомлення з описом вашої проблеми, і я зв'яжуся з вами найближчим часом!",
    contactsFormTitle:
      "Надішліть нам повідомлення з описом вашої проблеми, і ми зв'яжемося з вами найближчим часом!",
    contactsFormNameField: "Ім'я",
    contactsFormEmailField: "Ел. адреса",
    contactsFormMsgField: "Повідомлення",
    contactsFormSubmitField: "Відправити",
    contactsFormMsgLength: "Довжина повідомлення:",
  },
};

export default contactsFormData;
