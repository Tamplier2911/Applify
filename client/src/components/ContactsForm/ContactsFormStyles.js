import styled from "styled-components";

const getFont = props => {
  const { lang } = props;
  if (lang === "eng") {
    return `font-family: var(--font-title);`;
  } else if (lang === "rus" || lang === "ukr") {
    return `font-family: var(--font-cyrillic);`;
  }
  return `font-family: var(--font-title);`;
};

export const ContactsFormContainer = styled.div`
  display: grid;
  grid-row-gap: 4rem;
  grid-auto-rows: min-content;
`;

export const ContactsFormTitle = styled.h3`
  ${getFont}
  font-weight: 400;
  font-size: 1.7rem;
`;

export const ContactsFormItself = styled.form`
  display: grid;
  grid-row-gap: 2rem;
`;
