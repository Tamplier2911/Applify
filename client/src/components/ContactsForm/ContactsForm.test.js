import React from "react";

// component
import ContactsForm from "./ContactsForm";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<ContactsForm />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<ContactsForm store={store} />);
  });

  it("renders without an error", () => {
    const contactsForm = findByTestAttr(wrapper, "contacts-form");
    expect(contactsForm.length).toBe(1);
  });

  it("renders children - contacts form", () => {
    const contactsFormElement = findByTestAttr(
      wrapper,
      "contacts-form-element"
    );
    expect(contactsFormElement.length).toBe(1);
  });

  it("renders children - contacts form inputs", () => {
    const contactsFormInput = findByTestAttr(wrapper, "contacts-form-input");
    expect(contactsFormInput.length).toBe(3);
  });

  it("renders children - contacts form submit", () => {
    const contactsFormSubmit = findByTestAttr(wrapper, "contacts-form-submit");
    expect(contactsFormSubmit.length).toBe(1);
  });

  it("matches snapshot", () => {
    const contactsForm = findByTestAttr(wrapper, "contacts-form");
    expect(contactsForm).toMatchSnapshot();
  });
});
