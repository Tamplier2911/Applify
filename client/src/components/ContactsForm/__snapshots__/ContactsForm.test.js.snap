// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactsForm /> matches snapshot 1`] = `
<styled.div
  data-test="contacts-form"
>
  <styled.h3
    lang="eng"
  >
    Send us a message describing your problem, and we will get back to you shortly!
  </styled.h3>
  <styled.form
    autoComplete="off"
    data-test="contacts-form-element"
    onSubmit={[Function]}
  >
    <FormInput
      data-test="contacts-form-input"
      label="Name"
      name="name"
      onInputChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <FormInput
      data-test="contacts-form-input"
      label="Email"
      name="email"
      onInputChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <TextInput
      data-test="contacts-form-input"
      label="Message"
      max="2000"
      name="message"
      onInputChange={[Function]}
      required={true}
      rows="6"
      value=""
    />
    <Button
      data-test="contacts-form-submit"
      type="submit"
      value="Send"
    />
  </styled.form>
  <styled.p>
    Message length: 
    2000
  </styled.p>
</styled.div>
`;
